### ISSUE TITLE
Sample Project Loading Issue

### ISSUE THREAD

**Issue by mparrish91** [sonsongithub/reddift#206]
[http://i.imgur.com/lpRIaHY.gif](video)

Hi I downloaded the example project and everything seems to work well, but when I try to search for a listing nothing shows.  
Anything I should try?

My reddit app account info is added in the json file

Thanks for the great example btw

**Comment by sonsongithub** [comment]
Do you login with any reddit accounts?
I think search function may need a user authentication. You can login reddit.com left upper button.

**Comment by mparrish91** [comment]
have been struggling to get a user authenticated

i must be doing something wrong

[http://imgur.com/a/ZnY0P](link)

**Comment by sonsongithub** [comment]
It seems that your setting is almost good.
I guess, callback URL is no invalid? Could one use "+" in URL scheme?
If your URL does not have any problems, currently I do not have any suggestions...

**Comment by mparrish91** [comment]
removed the plus and made a new app on reddit site. seems to work. asked me to open in reddit.  after i accepted.  but the account didn't get added to the tableview

**Comment by mparrish91** [comment]
oh just noticed this in console

OSStatus error:[-25300] The specified item could not be found in the keychain.
OSStatus error:[-25300] The specified item could not be found in the keychain.
tokenIsNotAvailable

**Comment by sonsongithub** [comment]
Did you enable the application to use "Keychain Sharing" and "Data protection" in Xcode?

**Comment by mparrish91** [comment]
yeah the sample project came with that. this came up on google

http://stackoverflow.com/questions/16845169/error-when-trying-to-obtain-a-certificate-the-specified-item-could-not-be-found

I need to look into it more

**Comment by sonsongithub** [comment]
I got it.
Maybe, you should confirm that your tools can pass tests of the MiniKeychain or KeychainAccess.

**Comment by mparrish91** [comment]
Can I send you a zip of my project? I'm thinking I'm missing a library or
some code. Because I just readded my certificate.

Curious if it works in your machine

**Comment by sonsongithub** [comment]
I think that your problem is caused by Keychain.
When `func application(_ app: UIApplication, open url: URL, options: [UIApplicationOpenURLOptionsKey : Any] = [:])` is called, it's not caused by `reddift.framework` if the callback URL includes oAuth key.
Please check it.

**Comment by mparrish91** [comment]
https://www.dropbox.com/s/ql8z29atabl5k3c/reddift-master.zip?dl=0

**Comment by sonsongithub** [comment]
- You should use original application identifier, like "com.mparrish91.redditapp".
- You should use original keychain identifier, like "com.mparrish91.redditapp.keychain".
- You should enable to use "Data protection" on "Certificates, identifier and profiles" on iTunes connect.
- You should change identifier of URI Types from com.sonson.reddfit to other one.

**Comment by mparrish91** [comment]
thanks bro sorry to be a hassle with the project.  After you changed that
it started running?

**Comment by mparrish91** [comment]
Made all of those changes. checked data protection in iTunes connect too.
 still get that annoying error.  something must be up with my keychain

**Comment by sonsongithub** [comment]
I don't have much time to struggle with your project....
How about checking whether KeychainAccess example app runs on your environment correctly?
https://github.com/kishikawakatsumi/KeychainAccess

**Comment by mparrish91** [comment]
I just tried the example app and it runs fine on my computer.  Perhaps
something is wrong with my reddit project.  Going to re download and setup
everything.  Can I use pods to install the dependencies?

Can u send me a zip of ur current working version of ur reddit app?


Thank you again for your time

**Comment by sonsongithub** [comment]
My folder is as same as git repository.
You should check out it again, carefully. Recently,  at the latest commit, I changed structure of the reddift sample application source code.

```
git clone --recursive https://github.com/sonsongithub/reddift.git
carthage update --platform iOS
```

After above commands, check whether following file and folders exist at each path.
```
/framework/vendor/HTMLSpecialCharacters
/framework/vendor/MiniKeychain
/Carthage/Build/iOS/FLAnimatedImage.framework
/Carthage/Build/iOS/YouTubeGetVideoInfoAPIParser.framework
/Carthage/Build/iOS/UZTextView.framework
```

**Comment by mparrish91** [comment]
So I figured out my problem.

Step 1 was seeing if my redirect functions were getting called in the app
delegate. Then I slowly passed through each.

Noticed that the redirect URL scheme I get back from reddit is gifrecipes
but my redirect is gifRecipes and that is my name also.

So for a quick hack I changed the scheme in config file to lower case
gifrecipes and it worked.  username came in table and can now view
different subredits
